.ci/static-checks.sh

```diff
@@ -1102,6 +1103,132 @@ static_check_json()
 	done
 }
 
+# The dockerfile checker relies on the hadolint tool. This function handle its
+# installation if it is not found on PATH.
+# Note that we need a specific version of the tool as it seems to not have
+# backward/forward compatibility between versions.
+has_hadolint_or_install()
+{
+	# Global variable set by the caller. It might be overwritten here.
+	linter_cmd=${linter_cmd:-"hadolint"}
+	local linter_version=$(get_test_version "externals.hadolint.version")
+	local linter_url=$(get_test_version "externals.hadolint.url")
+	local linter_dest="${GOPATH}/bin/hadolint"
+
+	local has_linter=$(command -v "$linter_cmd")
+	if [[ -z "$has_linter" && "$KATA_DEV_MODE" == "yes" ]]; then
+		# Do not install if it is in development mode.
+		die "$linter_cmd command not found. You must have the version $linter_version installed to run this check."
+	elif [ -n "$has_linter" ]; then
+		# Check if the expected linter version
+		if $linter_cmd --version | grep -v "$linter_version" &>/dev/null; then
+			warn "$linter_cmd command found but not the required version $linter_version"
+			has_linter=""
+		fi
+	fi
+
+	if [ -z "$has_linter" ]; then
+		local download_url="${linter_url}/releases/download/v${linter_version}/hadolint-Linux-x86_64"
```

Comment:
Please use `uname -m` rather than hard-coding architecture names.

Reply:
btw, hadolint is only distributed to x86_64. Makes sense to skip the checker for non-x86 arches?

Reply:
That's unfortunate, but skipping seems like the best approach at this stage. I presume there checking those files on a single architecture will still find all the lint issues anyhow too.

---
Aside: If you haven't seen the logo for `hadolint`, this might just make your day :smile::

- https://github.com/hadolint/hadolint#haskell-dockerfile-linter

Let's keep our dockerfiles litter free and fragrant y'all!